### PR TITLE
ROX-18155: pg generic store: GetAll

### PR DESCRIPTION
**Pull request by janisz**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6766** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6766" target="_blank">Graphite</a> 
    * **PR #6767** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6767" target="_blank">Graphite</a> 
      * **PR #6768** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6768" target="_blank">Graphite</a> 
        * **PR #6769** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6769" target="_blank">Graphite</a>  👈
          * **PR #6770** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6770" target="_blank">Graphite</a> 
            * **PR #6771** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6771" target="_blank">Graphite</a> 
              * **PR #6772** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6772" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6769?utm_source=stack-comment).

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 1]
```diff
@@ -729,21 +729,6 @@ func (s *storeImpl) GetIDs(ctx context.Context) ([]{{$singlePK.Type}}, error) {
 }
 {{- end }}
 
-{{- if .GetAll }}
```
Comment (rhybrillou) [inline tools/generate-helpers/pg-table-bindings/store.go.tpl:732]: If not already done in the cleanup PR, make sure the get-all option is removed from the generator code as well as from the gen.go files that use it.

File: pkg/search/postgres/store.go [hunk 2]
```diff
@@ -180,3 +180,15 @@ func (s *GenericStore[T, PT]) Walk(ctx context.Context, fn func(obj PT) error) e
 	}
 	return nil
 }
+
+// GetAll retrieves all objects from the store.
+func (s *GenericStore[T, PT]) GetAll(ctx context.Context) ([]*T, error) {
+	defer s.setPostgresOperationDurationTime(time.Now(), ops.GetAll)
+
+	var objs []*T
```
Comment (rhybrillou) [inline pkg/search/postgres/store.go:188]: Why not `var objs []PT` and have `([]PT, error)` as return type ?

**Review by rhybrillou — APPROVED** [review]

File: tools/generate-helpers/pg-table-bindings/store.go.tpl (hunk 1 above)
Comment (rhybrillou) [reply tools/generate-helpers/pg-table-bindings/store.go.tpl:732]: The complete cleanup would impact:
- the gen.go files next to the stores generated by this pull request
- tools/generate-helpers/pg-table-bindings/migration.go.tpl (or the file could be removed completely, as well as the gen.go file in the migrator)
- tools/generate-helpers/pg-table-bindings/store.go.tpl (but I think the current pull request already removes all occurrences)
- tools/generate-helpers/pg-table-bindings/store_test.go.tpl (to let all stores test the GetAll function)
- tools/generate-helpers/pg-table-bindings/main.go

Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/store.go.tpl:730]: So you think we should expose `GetAll` method to all stores? I wander what was the reason to have it as optin

File: pkg/search/postgres/store.go (hunk 2 above)
Comment (janisz) [reply pkg/search/postgres/store.go:185]: Indeed this will save us a cast

File: tools/generate-helpers/pg-table-bindings/store.go.tpl (hunk 1 above)
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/store.go.tpl:730]: https://issues.redhat.com/browse/ROX-18312

**Comment by janisz** [comment]
@janisz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6769).

**Comment by janisz** [comment]
[Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6769) rebased this pull request as part of a merge.

**Comment by janisz** [comment]
@janisz merged this pull request with [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6769).

**Pull request merged**